lib/changes.js

```diff
@@ -0,0 +1,102 @@
+var nanoiterator = require('nanoiterator')
+var inherits = require('inherits')
+
+module.exports = Iterator
+
+function Iterator (multi) {
+  if (!(this instanceof Iterator)) return new Iterator(multi)
+  nanoiterator.call(this)
+
+  this._multi = multi
+  this._end = []
+  this._nodes = []
+}
+
+inherits(Iterator, nanoiterator)
+
+Iterator.prototype._open = function (cb) {
+  var self = this
+
+  this._multi.heads(function (err, heads) {
+    if (err) return cb(err)
+
+    var writers = self._multi._writers
+
+    for (var i = 0; i < writers.length; i++) {
+      self._end.push(highestClock(heads, i))
+      self._nodes.push(null)
+    }
+
+    self._updateAll(cb)
+  })
+}
+
+Iterator.prototype._updateAll = function (cb) {
+  var self = this
+  var missing = 0
+  var error = null
+  var writers = this._multi._writers
+
+  for (var i = 0; i < this._nodes.length; i++) {
+    if (this._end[i] && !this._nodes[i]) {
+      missing++
+      writers[i].get(0, onnode)
+    }
+  }
+
+  if (!missing) cb(null)
```

Comment:
`process.nextTick(cb, null)`?

Reply:
this is guaranteed to run in the nexttick already so no need :)